--- a/const.py
+++ b/const.py
@@ -45,7 +45,10 @@
 DEFAULT_PROBE_METHODS = (
     (ApplicationType.GECKO_BOOTLOADER, 115200),
     (ApplicationType.EZSP, 115200),
+    (ApplicationType.EZSP, 230400),
     (ApplicationType.EZSP, 460800),
+    (ApplicationType.SPINEL, 115200),
+    (ApplicationType.SPINEL, 230400),
     (ApplicationType.SPINEL, 460800),
     (ApplicationType.CPC, 460800),
     (ApplicationType.CPC, 115200),
